src/beast/beast/nudb/detail/field.h

```diff
@@ -103,25 +112,25 @@ read (istream& is, U& u)
 template <class T, class U, std::enable_if_t<
     std::is_same<T, uint24_t>::value>* = nullptr>
 void
-read (istream& is, U& u)
+readp (void const* v, U& u)
 {
-    T t;
     std::uint8_t const* p =
-        is.data(field<T>::size);
-    t = (T(*p++)<<16) | t;
-    t = (T(*p++)<< 8) | t;
-    t =  T(*p  )      | t;
+        reinterpret_cast<std::uint8_t const*>(v);
+    std::uint32_t t;
```

Comment:
Does it ever... or the next line needs to lose the `| t` part. Nice catch @seelabs!